### ISSUE TITLE
Issue with hebo/acq_optimizers/evolution_optimizer.py 

### ISSUE THREAD

**Issue by Sunny276** [huawei-noah/HEBO#73]
When the eval() function in the acquisition function class involves gradient backward, because line 102 of hebo/acq_optimizers/evolution_optimizer.py is ''with torch.no_grad():'', the gradient cannot be passed back. delete this line solves the problem. Why add this line ? Will deleting this line affect the final output result ?

background: I am using the genetic algrithm to find good hyperparameters for neural network training using pytorch.

**Comment by AntGro** [comment]
It is there as we assume that if you optimize the acquisition function with the evolution optimizer you don't optimize it with gradient descent.

**Comment by Sunny276** [comment]
Okay, I got it. Thank you for your response, and I understand your point. However, selecting appropriate hyperparameters for neural networks is a common optimization problem. If this function is not designed to handle that, I would appreciate it if this could be clearly mentioned in the documentation to avoid any misunderstandings. This is particularly important because the HEBO usage guide suggests that it is capable of optimizing neural network hyperparameters.
![Screenshot 2024-08-31 at 12 24 14](https://github.com/user-attachments/assets/06321ada-a4ca-4c15-a3f8-a9669900cc5c)

**Comment by AntGro** [comment]
HEBO won the [NeurIPS 2020 Black-Box Optimisation Challenge for Machine Learning](https://bbochallenge.com/leaderboard) so it is of course compatible with neural net hyperparameter optimization. Yo need gradient when you evaluate the black-box, I.e. when you train your model with the suggested hyper parameters and get a validation loss, but you don’t need to use gradient when HEBO internally optimizes its acquisition function to find the next set of hyperparameters to try.
